### ISSUE TITLE
Auras are not removed if the same minion type is in the valid target list

### ISSUE THREAD

**Issue by beheh** [jleclanche/fireplace#108]
The following test fails whenever the marked line is enabled:

```
def test_mechwarper():
	game = prepare_game()
#      game.player1.give("GVG_082") # this line breaks the test
	mechwarper = game.player1.give("GVG_006")
	goldshire = game.player1.give(GOLDSHIRE_FOOTMAN)
	harvest = game.player1.give("EX1_556")
	clockwork = game.player1.give("GVG_082")
	assert harvest.cost == 3
	assert goldshire.cost == 1
	assert clockwork.cost == 1

	mechwarper.play()
	assert harvest.cost == 3 - 1
	assert goldshire.cost == 1
	assert clockwork.cost == 0

	clockwork.play()
	assert clockwork.cost == 1

	game.current_player.give(SILENCE).play(target=mechwarper)
	assert harvest.cost == 3
	assert goldshire.cost == 1
	assert clockwork.cost == 1

	mechwarper.destroy()
	assert harvest.cost == 3
	assert goldshire.cost == 1
	assert clockwork.cost == 1
```

**Comment by beheh** [comment]
See beheh/fireplace@df5d474dc40dc6746ec386a598a142ca70a2deea.

**Comment by beheh** [comment]
Fixed by 030b338d093ec19dbf3bdf9fe93a386a0c69e656...8404c339f14f56d8b2664f652fe3cba024e6f4cf.
